impl/client/src/main/javascript/web/pentaho/visual/views/echarts/Abstract.js

```diff
@@ -256,10 +271,94 @@ define([
 
       this._configureOptions();
 
+      this._chart.on("dblclick", function (params) {
+        this._onExecute(params);
+      }, this);
+
+      // When 'click' and 'dblclick' are bound to same element
+      // and user double-clicks, 'click' is fired twice followed by dbl-click
+      // Reference: https://stackoverflow.com/a/7119911
+      var clickTimeout;
+      this._chart.on("click", function (params) {
+        clearTimeout(clickTimeout);
+        var me = this;
+        clickTimeout = setTimeout(function() {
+          me._onSelect(params);
+        }, 300);
+      }, this);
+
       // Draw the chart
       this._chart.setOption(this._echartOptions);
     },
 
+    _onExecute: function(params) {
+      params.cancelBubble = true;
+      var model = this.model;
+      if(model.isDirty) {
+        return;
+      }
+
+      var filter = null;
```

Comment:
Variable is declared twice. Same for `_onSelect` method.

Reply:
Will make the change.